nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/work/BlobListZipWork.java

```diff
@@ -114,11 +124,35 @@ protected void updateAndCompleteStoreEntry(List<Blob> blobs) {
     @Override
     public void work() {
         openUserSession();
+        List<Blob> blobList = new ArrayList<Blob>();
+        DownloadService downloadService = Framework.getService(DownloadService.class);
+        for (String docId : docIds) {
+            DocumentRef docRef = new IdRef(docId);
+            if (!session.exists(docRef)) {
+                log.info(String.format("Cannot retrieve document '%s', probably deleted in the meanwhile", docId));
+                continue;
+            }
+            DocumentModel doc = session.getDocument(docRef);
+            Blob blob = downloadService.resolveBlob(doc);
+            if (blob == null) {
+                log.trace("Not able to resolve blob");
+                continue;
+            } else if (!downloadService.checkPermission(doc, null, blob, "download", Collections.emptyMap())) {
+                log.debug(String.format("Not allowed to bulk download blob for document %s", doc.getId()));
```

Comment:
Can surround with `if (log.isDebugEnabled())` to avoid the unnecessary call to `String.format` and maybe add `doc.getPathAsString()` along with the doc id in the message to ease debugging.